### ISSUE TITLE
GIT_VERSION outside of version controlled source code area

### ISSUE THREAD

**Issue by davidlange6** [bkryza/clang-uml#211]
I ran into this problem when using a released version (or a tar ball without a .git area..) of clang-um:

```
CMake Error at cmake/GitVersion.cmake:20 (string):
  string sub-command REGEX, mode MATCH needs at least 5 arguments total to
  command.
Call Stack (most recent call first):
  CMakeLists.txt:58 (setup_git_version)
```

It looks like the default here  is insufficient (better if this could be autofilled with the released version but that is not a blocker?)

https://github.com/bkryza/clang-uml/blob/f3a0d98ee6fd0fffb4fa2a255f0c337f6d81db9d/cmake/GitVersion.cmake#L17 

I could work around this by adding something like
```
  -DGIT_VERSION="0.4.1-9-g205d6de" \
```
to my cmake command

**Comment by bkryza** [comment]
@davidlange6 Thanks, I've fixed the CMake macro so that it doesn't error, but it does set the version to `"0.0.0-unknown"` as currently I don't have a better idea how to set the version properly in the released tarball outside of Git. 

Setting it manually in the source is not an option as I will always forget it ;-)

Also, if you download tarball from GitHub for the latest `master` branch, it will not necessarily correspond to the latest release, so it would be confusing, which version is used.

So basically if you download a tarball from GitHub without `.git`, you have to provide the version using `GIT_VERSION` variable...
